config/routes.rb

```diff
@@ -19,17 +19,42 @@
   }
   get '/v1/(*path)', to: redirect { |_params, request| "/api#{request.path}?#{request.query_string}" }
   get '/v2/(*path)', to: redirect { |_params, request| "/api#{request.path}?#{request.query_string}" }
+
   get '/api/:version/commodities/:id', constraints: { id: /\d{2}00000000/ }, to: redirect { |_params, request|
     path = request.path.gsub('commodities', 'chapters').gsub('00000000', '')
-    "https://#{ENV['HOST']}#{path}"
+    query = URI(request.url).query
+
+    url = "https://#{ENV['HOST']}#{path}"
+
+    if query
+      "#{url}?#{query}" if query
```

Comment:
You have already checked `query`, in the line above
it  just seems to be equal to:

`query ? "#{url}?#{query}" : url`